rotkehlchen/tests/db/test_db_upgrades.py

```diff
@@ -2718,6 +2722,35 @@ def test_upgrade_db_45_to_46(user_data_dir: 'Path', messages_aggregator):
 @pytest.mark.parametrize('use_clean_caching_directory', [True])
 def test_upgrade_db_46_to_47(user_data_dir, messages_aggregator):
     """Test upgrading the DB from version 46 to version 47"""
+
+    def _check_tokens_existence(user_db_cursor: 'DBCursor', expect_removed: bool = False):
+        """Assert whether the tokens are in the db or not depending on `expect_removed`"""
+        for table, column, count in [
+            ('assets', 'identifier', 2),
+            ('history_events', 'asset', 1),
+            ('timed_balances', 'currency', 1),
+            ('evm_accounts_details', 'value', 2),
+            ('manually_tracked_balances', 'asset', 1),
+        ]:
+            assert user_db_cursor.execute(
+                f'SELECT COUNT(*) FROM {table} WHERE {column} IN (?, ?)',
+                (uniswap_erc20_token.identifier, uniswap_erc721_token.identifier),
+            ).fetchone()[0] == 0 if expect_removed else count
+
+        with GlobalDBHandler().conn.read_ctx() as global_db_cursor:
+            assert global_db_cursor.execute(
+                'SELECT COUNT(*) FROM assets WHERE identifier IN (?, ?)',
+                (uniswap_erc20_token.identifier, uniswap_erc721_token.identifier),
+            ).fetchone()[0] == 0 if expect_removed else 2
+
+        if expect_removed:
+            # TODO: the history_events entry will need removed from this assertion
```

Comment:
mark it as custom. Then the test will be even more accurate and you won't need to fix this test after resetting of events.